src/server/lib/locale-handler.js

```diff
@@ -0,0 +1,18 @@
+import * as cookie from '../lib/cookie'
+
+/**
+ * Get locale based on body / query param / cookie
+ */
+export default function localeHandler (req, res, cookieOptions) {
+  const { body, query } = req
+  // get locale from body, query or cookie
+  const localeCookieValue = req.cookies[cookieOptions.locale.name] || null
+  const locale = body.locale || query.locale || localeCookieValue || null
```

Comment:
There should never be a scenario where this would fallback all the way to null, unless the app is single-locale. Can we even access the next config js file? I am not that familiar with the inner workings of NextJS 😄